<a href="https://colab.research.google.com/github/rn-byte/Cache-Api-Data-Flutter/blob/main/Data_Science_Expo.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#So Right Now i have Selected SALARY_PREDICTION dataset For The EXPO. From that dataset we need to find the pattern ,trend and make prediction.
**The Salary Dataset contains four column :**
 **a.YearsExperience**
 **b. EducationLevel**
 **c. Age**
 **d. Salary**
**and 200 rows**

**1. First lets import necessary libraries**


In [2]:
# Imoirting Libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
print('Libraries Imported')

Libraries Imported


**2. Lets Load and Read the Data Set i.e  [salary_prediction.csv](https://drive.google.com/file/d/1mslWg8xmZYrk5gqFpWaOLtYVxuG_3W5N/view)**

In [3]:
#Loading and Reading CSV File
df=pd.read_csv('/content/salary_prediction.csv')

**Displaying First and Last 10 Data**

In [4]:
#Displaing/ printing First 10 Data
df

YearsExperience  EducationLevel  Age        Salary
0                7.5               4   51  54313.592802
1               19.0               4   49  82663.845282
2               14.6               3   54  72196.732571
3               12.0               1   59  60463.694565
4                3.1               4   35  35756.199258
..               ...             ...  ...           ...
195              7.0               4   32  48672.104217
196             14.5               1   47  59588.013310
197             17.9               3   39  74440.615169
198             17.7               4   47  80061.968880
199             15.6               2   27  60037.209061

[200 rows x 4 columns]

In [5]:
#Displaying Last 10 Data
df.tail()

YearsExperience  EducationLevel  Age        Salary
195              7.0               4   32  48672.104217
196             14.5               1   47  59588.013310
197             17.9               3   39  74440.615169
198             17.7               4   47  80061.968880
199             15.6               2   27  60037.209061

**2. Checking DataSet INFORMATION**

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 200 entries, 0 to 199
Data columns (total 4 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   YearsExperience  200 non-null    float64
 1   EducationLevel   200 non-null    int64  
 2   Age              200 non-null    int64  
 3   Salary           190 non-null    float64
dtypes: float64(2), int64(2)
memory usage: 6.4 KB


**So the salary Column contains 10 null value.**

**3. Describing DataSet Information.**

In [7]:
df.describe()

YearsExperience  EducationLevel         Age         Salary
count       200.000000      200.000000  200.000000     190.000000
mean          9.678000        2.555000   40.555000   53284.165839
std           5.896913        1.141706   11.376029   31801.194434
min           0.100000        1.000000   20.000000    6127.861706
25%           4.600000        2.000000   31.000000   32182.506476
50%           9.900000        2.500000   42.000000   50689.864885
75%          15.125000        4.000000   50.000000   66201.111163
max          19.700000        4.000000   59.000000  269683.212127

**This table provides descriptive statistics for four variables: YearsExperience, EducationLevel, Age, and Salary**

**Overall:**

**count**: This shows the number of observations for each variable. We have 200 observations for YearsExperience, EducationLevel, and Age, but only 190 for Salary. This suggests there are 10 missing values for Salary.

**mean:** This is the average value for each variable. For example, the average YearsExperience is 9.678 years, and the average Salary is approximately $53,284.

**std:** This is the standard deviation, which measures the spread or dispersion of the data around the mean. A higher standard deviation indicates greater variability. Salary has a relatively high standard deviation, suggesting a wide range of salaries in the dataset.

**Specifics:**

**YearsExperience:** Ranges from 0.1 years to 19.7 years, with an average of 9.678 years.

**EducationLevel:** This appears to be a categorical variable represented numerically. Without more context, it's hard to say for sure what each level represents (e.g., 1 = High School, 2 = Bachelor's, etc.). However, we can see that the average is 2.555, and the values range from 1 to 4.

**Age:** Ranges from 20 to 59 years old, with an average age of 40.555 years.

**Salary:** Ranges from approximately $6,128$ to $269,683$, with an average of $53,284. The large range and high standard deviation suggest potential outliers or a skewed distribution.

**Quartiles:**

**The 25th, 50th (median), and 75th percentiles give us a sense of the distribution:**

**YearsExperience:** 25% of individuals have less than 4.6 years of experience, 50% have less than 9.9 years, and 75% have less than 15.125 years.

**Age:** 25% of individuals are younger than 31,

50% are younger than 42,

 and 75% are younger than 50.

**Salary:**

25% of individuals earn less than $32,182,

50% earn less than $50,690 and

75% earn less than $66,201.

**Standard deviation is a measure of uncertainty.**
**A low standard deviation means that most of the numbers are close to the mean (average) value.
A high standard deviation means that the values are spread out over a wider range.**

**4. Checking The Shape of DATA Set.**

In [44]:
#It will show output like (row,column)
df.shape

(200, 4)

In [45]:
df

YearsExperience     EducationLevel  Age        Salary
0                7.5                PhD   51  54313.592802
1               19.0                PhD   49  82663.845282
2               14.6    Master's Degree   54  72196.732571
3               12.0        High School   59  60463.694565
4                3.1                PhD   35  35756.199258
..               ...                ...  ...           ...
195              7.0                PhD   32  48672.104217
196             14.5        High School   47  59588.013310
197             17.9    Master's Degree   39  74440.615169
198             17.7                PhD   47  80061.968880
199             15.6  Bachelor's Degree   27  60037.209061

[200 rows x 4 columns]

In [46]:
# Your code here
#Checking for Null Values
df.isnull().sum()

YearsExperience     0
EducationLevel      0
Age                 0
Salary             10
dtype: int64

***Usings linear or polynomial interpolation to estimate missing values based on trends in the dataset.***

In [47]:
df['Salary'] = df['Salary'].interpolate(method='linear')

In [48]:
df.isnull().sum()

YearsExperience    0
EducationLevel     0
Age                0
Salary             0
dtype: int64

                 YearsExperience  EducationLevel       Age    Salary
YearsExperience         1.000000        0.059743  0.152809  0.686562
EducationLevel          0.059743        1.000000 -0.068329  0.180782
Age                     0.152809       -0.068329  1.000000  0.092454
Salary                  0.686562        0.180782  0.092454  1.000000
